fix: uppercase in path

Jannchie opened the pull request:
<!---
☝️ PR title should follow conventional commits (https://conventionalcommits.org)
-->

### 🔗 Linked issue

close #2162 

### ❓ Type of change

<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

<!-- Describe your changes in detail -->
<!-- Why is this change required? What problem does it solve? -->
<!-- If it resolves an open issue, please link to the issue here. For example "Resolves #1337" -->

Currently, if we construct a path like /zh-CN/foo.md, the corresponding document cannot be found. This is because the path is automatically converted to lowercase. I think the existence of such a path is reasonable and should not be converted. I don't understand why this is done, so in order not to introduce destructive changes, I added a flag named keepUppercase to allow uppercase letters in the path.

I wasn't sure where this flag should be placed, so I tentatively put it under the experimental options.

Resolves #2162 

### 📝 Checklist

<!-- Put an `x` in all the boxes that apply. -->
<!-- If your change requires a documentation PR, please link it appropriately -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

nobkd commented:
Can you add this option to the docs too?
I think this is the right place: https://github.com/nuxt/content/blob/main/docs/content/4.api/3.configuration.md

Also, I don't really think, that this is a breaking change. So I think that it does not have to be in the experimental features. What do you think?

Jannchie commented:
> Also, I don't really think, that this is a breaking change. So I think that it does not have to be in the experimental features. What do you think?

You are right, I moved the location of this option and added the documentation! Thanks for the continued follow up.

Barbapapazes commented on src/module.ts:
```diff
@@ -220,6 +220,12 @@ export interface ModuleOptions {
+   *
+   * @default false
+   */
+  keepUppercase: boolean
```
This name may be too vague since we do not understand it's related to path. 

Maybe something like respectPathCase, when false, it lower case or respectFilenameCase

Jannchie replied:
I've rename the name to `respectPathCase`.

nobkd commented on docs/content/4.api/3.configuration.md:
```diff
@@ -410,3 +410,10 @@ Toggles the document-driven mode.
+
+## `respectPathCase`
+
+- Type: `Boolean`{lang=ts}
```
```suggestion
- Type: `boolean`{lang=ts}
```
Currently making types in the docs lowercase. See #2167

Jannchie replied:
fixed..!

Barbapapazes commented on docs/content/4.api/3.configuration.md:
```diff
@@ -410,3 +410,10 @@ Toggles the document-driven mode.
 | `injectPage`      |      `boolean`      | Inject `[...slug].vue` pre-configured page                     |
 | `trailingSlash`   |      `boolean`      | Add a slash at the end of  `canonical` and `og:url`            |
+
+## `respectPathCase`
```
Maybe you could provide an example to help dev to understand types of issues that this options can solve

Jannchie replied:
Added!

Barbapapazes replied:
Perfect 👌

Barbapapazes commented:
Nice PR!

farnabaz reviewed (approved):
Perfect 👍 
Thanks ❤️

farnabaz commented on src/runtime/transformers/path-meta.ts:
```diff
@@ -67,7 +67,8 @@ const isPartial = (path: string): boolean => path.split(/[:/]/).some(part => par
  */
 export const generatePath = (path: string, { forceLeadingSlash = true } = {}): string => {
-  path = path.split('/').map(part => slugify(refineUrlPart(part), { lower: true })).join('/')
+  const { content } = useRuntimeConfig().public
```
We should avoid using Nuxt generated helpers (like `useRuntimeConfig`) in transformers. Transformers and in general the content parser could be able to work outside of Nuxt Context.

There is already several usecases of parser outside of Nuxt.

`respectPathCase`  could be passed in second argumant of this function and transformer itseld
